sark/sark_event.c

```diff
@@ -390,43 +406,35 @@ void event_run(uint restart)
     }
 }
 
+static inline event_t *take_one_event_from_queue(proc_queue_t *queue) {
+    uint cpsr = cpu_int_disable();  // Interrupts off to manipulate queue
+    event_t* e = queue->proc_head;  // Get head of queue
+    if (e != NULL) {
+	queue->proc_head = e->next; // Remove from queue
+    }
+    cpu_int_restore(cpsr);
+    e->next = NULL;                 // Value here is irrelevant to caller
```

Comment:
[Line 416] what is the effect of this statement if e == NULL?

Reply:
Good catch.

Reply:
That line should have been two lines higher, inside the not-NULL-guarded section. (Strictly it doesn't need to be inside the disabled-interrupts section, but it's better to make the code obviously correct.)

Fixed.